src/services/index.ts

```diff
@@ -21,10 +21,10 @@ export const startUpdateAnalysisPipeline = function (): Promise<void> {
 export const handleData = new Writable({
   objectMode: true,
   write(source: TsvColumns, _encoding, callback) {
-    getCacheByKey(`sample:${source.specimen_collector_sample_ID}`)
+    getCacheByKey(`${source.study_id}:${source.specimen_collector_sample_ID}`)
```

Comment:
should there be a utility method for building this key that we use both when calling save and get?

Reply:
followed your suggestion, I then added a method to take care for Hash key format in a single place. Thanks.